### PR TITLE
Implement BanIP and UnbanIP commands

### PR DESCRIPTION
**Pull request by aeltoum**
[https://trello.com/c/xj9MzDro](url)

-Implemented BanIP and UnbanIP alongside Error handling and logging.

### CODE REVIEW

File: Backend/Models/Messages/CommandMessage.cs [hunk 1]
```diff
@@ -372,6 +374,109 @@ private void ProcessUnban()
...
+            {
+                CLogger.Error("Command not invoked: User cannot ban themself");
+                SendRefuse("You can't ban yourself idiot");
+                return;
+            }
+
+            if (BannedIp.AlreadyExists(ipToBan))
+            {
+                CLogger.Error("Command not invoked: User is already banned");
+                SendRefuse("User is already banned");
```
Comment (ahmed-okkio) [inline Backend/Models/Messages/CommandMessage.cs]: IP not user in both error messages

File: Backend/Models/Users/BannedIp.cs [hunk 2]
```diff
@@ -25,9 +28,22 @@ public void SaveToDb()
...
         {
-            List<BannedIp> ipsFromDb = context.BannedIps.Where(m => m.IpAddress == IpAddress).ToList();
+            context.Remove(this);
+            context.SaveChanges();
+        }
+
+        public static BannedIp? GetBannedIpFromDb(string bannedIp)
+        {
+            List<BannedIp> bannedIpsFromDb = context.BannedIps.Where(m => m.IpAddress == bannedIp).ToList();
+            if (bannedIpsFromDb.Count == 0) return null;
```
Comment (ahmed-okkio) [inline Backend/Models/Users/BannedIp.cs]: wrap the return in brackets please, we're not using single line ifs

File: Backend/Models/Users/UserManager.cs [hunk 3]
```diff
@@ -41,20 +41,16 @@ public static void Unban(User user)
...
+        public static void BanIp(string ip, string reason = "")
         { 
-            BannedIp ipToBan = new BannedIp(ip);
-            if (ipToBan.AlreadyExists())
-            {
-                CLogger.Error($"Attempt to ban IP {ip} failed: IP already banned");
-                return;
-            }
-
+            BannedIp ipToBan = new BannedIp(ip,reason);
```
Comment (ahmed-okkio) [inline Backend/Models/Users/UserManager.cs:46]: Should we disconnect all users with a matching IP?

**Comment by ahmed-okkio** [comment]
Where do we prevent banned IP's from connecting ?

File: Backend/Models/Users/BannedIp.cs (hunk 2 above)
Comment (aeltoum) [reply Backend/Models/Users/BannedIp.cs]: Not even my code smh

**Review by ahmed-okkio — APPROVED** [review]

**Pull request merged**